ts_src/confidential.ts

```diff
@@ -116,10 +116,36 @@ export async function rangeProof(
   minValue?: string,
   exp?: number,
   minBits?: number,
+): Promise<Buffer> {
+  const nonce = await nonceHash(blindingPubkey, ephemeralPrivkey);
+  return rangeProofWithoutNonceHash(
+    value,
+    nonce,
+    asset,
+    assetBlindingFactor,
+    valueBlindFactor,
+    valueCommit,
+    scriptPubkey,
+    minValue,
+    exp,
+    minBits,
+  );
+}
+
+export async function rangeProofWithoutNonceHash(
```

Comment:
why do you need this method? I don't think the rangeproof is valid if built without the ECDH nonce!

EDIT: Ok, after reviewing also the blinder I got the purpose of this function. Instead of declaring a rangeProofWithoutNonce func, following the example of go-elements we could have a rangeProof method that expects a nonce and 2 wrappers that will call it: one for helping creating the nonce rangeProofWithNonce (essentially a renamaing of the current method) and one for helping doing the same with a key rangeProofWithKey (helpfull for issuance blinding).